source/code/source/playfab/PlayFabEventBuffer.cpp

```diff
@@ -57,7 +57,7 @@ namespace PlayFab
         }
     }
 
-    PlayFabEventBuffer::EventProducingResult PlayFabEventBuffer::TryPut(std::shared_ptr<const IPlayFabEmitEventRequest> request) // This must be thread-safe
+    PlayFabEventBuffer::EventProducingResult PlayFabEventBuffer::TryPut(const std::shared_ptr<const IPlayFabEmitEventRequest>& request) // This must be thread-safe
```

Comment:
disagree

Reply:
I tried changing this function signature to support a unique_ptr, but it causes some issues that would essentially lead to a larger refactor. So I'll accept the disagreement here too.